substrate/frame/referenda/src/types.rs

```diff
@@ -144,15 +148,18 @@ pub trait TracksInfo<Balance, Moment> {
 	/// The origin type from which a track is implied.
 	type RuntimeOrigin;
 
-	/// Return the array of known tracks and their information.
-	fn tracks() -> &'static [(Self::Id, TrackInfo<Balance, Moment>)];
+	/// The iterator returned by `Self::tracks()` to lazily traverse the available tracks
+	type TracksIter: Iterator<Item = (Self::Id, TrackInfo<Balance, Moment>)>;
```

Comment:
Having something that may returns an iterator or slice would may be better

Reply:
Mmm you mean return something like `Either<&'static [(_, _)], Self::TracksIter>`? it feels a bit ugly for users to handle the variants compared to this more natural way. Let me try something like returning an `IntoIterator` there the `Item` can be either an owned or a borrowed value to not lose the benefit of not having to clone/move the `TrackInfo`